util/argconfig.c

```diff
@@ -619,3 +619,49 @@ bool argconfig_parse_seen(struct argconfig_commandline_options *s,
 
 	return false;
 }
+
+struct argconfig_commandline_options *argconfig_merge(struct argconfig_commandline_options *opts1,
+						      struct argconfig_commandline_options *opts2)
+{
+	struct argconfig_commandline_options *opts;
+	int i;
+	int num1 = 0;
+	int num2 = 0;
+
+	for (i = 0; opts1[i].option; i++)
+		num1++;
+
+	for (i = 0; opts2[i].option; i++)
+		num2++;
+
+	opts = calloc(1, sizeof(*opts) * (num1 + num2 + 1));
+	if (!opts) {
+		perror("Failed to allocate memory for opts");
+		return NULL;
+	}
+
+	memcpy(opts, opts1, sizeof(*opts) * num1);
+	memcpy(&opts[num1], opts2, sizeof(*opts) * num2);
+
+	return opts;
+}
+
+int argconfig_parse_merge(int argc, char *argv[], const char *program_desc,
```

Comment:
I would keep the name `argconfig_parse()` and rename the function above to something like `option_merge()` or so. This avoids that all the callsides get a new function name. The option merge aspect is just an internal thing.

Reply:
Ah I see we still have `argconfig_parse()` users left. Maybe in this case just introduce a `argconfig_nvmf_parse()` function which adds the default options automatically?

Reply:
And now I see why you left the define :) It's because it depends on local variables of the calling context. IMO, this is a bit ugly and I think it would be better to use a `struct nvmf_args` and hand it into the parser.